gef.py

```diff
@@ -95,13 +96,18 @@
     # Compat Py2/3 hacks
     def range(*args):
         """Replace range() builtin with an iterator version."""
-        if len(args) < 1:
-            raise TypeError()
-        start, end, step = 0, args[0], 1
-        if len(args) == 2: start, end = args
-        if len(args) == 3: start, end, step = args
+        if len(args) == 0:
+            raise TypeError('range expected 1 arguments, got 0')
+        if len(args) >= 4:
+            raise TypeError('range expected at most 3 arguments, got 4')
```

Comment:
No need to hard code `4`, this will show the same error message if we pass 5 args.
```python
if len(args) > 3:
   raise TypeError('range expected at most 3 arguments, got {}'.format(len(args)))
```
is more generic.